QuickOSM/core/process.py

```diff
@@ -132,19 +132,30 @@ def open_file(
                 }
                 memory_layer = item['vector_layer']
 
-                encoding = get_default_encoding()
-                writer = QgsVectorFileWriter(
-                    outputs[layer],
-                    encoding,
-                    memory_layer.fields(),
-                    osm_geometries[layer],
-                    memory_layer.crs(),
-                    output_format)
-
-                for f in memory_layer.getFeatures():
-                    writer.addFeature(f)
-
-                del writer
+                if output_format.driver_name == "GPKG":
+                    options = QgsVectorFileWriter.SaveVectorOptions()
+                    layer_name = layer + "_name"
+                    options.layerName = final_layer_name
+                    if os.path.exists(outputs[layer]):
+                        options.actionOnExistingFile = QgsVectorFileWriter.AppendToLayerAddFields
+                    else:
+                        options.actionOnExistingFile = QgsVectorFileWriter.CreateOrOverwriteFile
+                    QgsVectorFileWriter.writeAsVectorFormat(
+                        memory_layer, outputs[layer_name], options)
+                else:
+                    encoding = get_default_encoding()
+                    writer = QgsVectorFileWriter(
```

Comment:
Can you switch to give the layer in the static function ?
It would be more efficient.

Also, can you use V2 and V3 ? (according to QGIS version).
It will make it easier to switch/clean for QGIS 4.0 when the function will be removed.

Reply:
I tried V2 and V3 but qgis didn't recognize the functions for some reason